Library/Homebrew/utils/shfmt.sh

```diff
@@ -56,4 +69,155 @@ then
   exit
 fi
 
-"${SHFMT}" "${SHFMT_ARGS[@]}" "${FILES[@]}"
+
+check_read_and_write() {
+  [[ -f "$1" && -w "$1" ]]
+}
+
+###
+### Custom shell script styling
+###
+
+check_multiline_for_statements() {
+  local file="$1"
+  check_read_and_write "${file}" || return 1
+
+  # check pattern:
+  # for var in ... \
+  #            ...; do
+
+  # use the following instead (keep for statements only one line)
+  # ARRAY=(
+  #   ...
+  #   ...
+  # )
+  # for var in "${ARRAY[@]}"
+  # do
+
+  # TODO: use bash built-in regex match syntax instead
+  if grep -qE '^\s*for .*\\\(#.*\)\?$' "${file}"
+  then
+    # TODO: add line number
+    onoe "Multi-line for statement detected."
+    return 1
+  fi
+}
+
+check_IFS_newline() {
+  local file="$1"
+  check_read_and_write "${file}" || return 1
+
+  # check pattern:
+  # IFS=$'\n'
+  #
+  # use the following instead
+  # while read -r line
+  # do
+  # done < <(command)
+
+  # TODO: use bash built-in regex match syntax instead
+  if grep -qE "^[^#]*IFS=\\\$'\\\\n'" "${file}"
+  then
+    # TODO: add line number
+    onoe "Pattern \`IFS=\$'\\\\n'\` detected."
+    return 1
+  fi
+}
+
+wrap_then() {
+  local file="$1"
+  check_read_and_write "${file}" || return 1
+
+  # TODO: wrap `then` to a separated line
+  # before:                   after:
+  # if [[ ... ]]; then        if [[ ... ]]
+  #                           then
+  #
+  # before:                   after:
+  # if [[ ... ]] ||           if [[ ... ]] ||
+  #   [[ ... ]]; then           [[ ... ]]
+  #                           then
+
+  true
+}
+
+# Probably merge into the above function
+wrap_do() {
+  local file="$1"
+  check_read_and_write "${file}" || return 1
+
+  # TODO: wrap `do` to a separated line
+  # before:                   after:
+  # for var in ...; do        for var in ...do
+  #                           do
+
+  true
+}
+
+align_multiline_if_condition() {
+  local file="$1"
+  check_read_and_write "${file}" || return 1
+
+  # TODO: align multiline if condition (indent with 3 spaces or 6 spaces (start with "-"))
+  # before:                   after:
+  # if [[ ... ]] ||           if [[ ... ]] ||
+  #   [[ ... ]]                  [[ ... ]]
+  # then                      then
+  #
+  # before:                   after:
+  # if [[ -n ... || \           if [[ -n ... || \
+  #   -n ... ]]                       -n ... ]]
+  # then                        then
+
+  true
+}
+
+# TODO: it's hard to align multiline switch cases
+align_multiline_switch_cases() {
+  true
+}
```

Comment:
It's probably hard to tweak `shfmt` results for switch case statements.See example https://github.com/Homebrew/brew/pull/12044#issuecomment-918251901 here.